activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -950,6 +950,7 @@ def foreign_key_column_for(table_name) # :nodoc:
 
       def foreign_key_options(from_table, to_table, options) # :nodoc:
         options = options.dup
+        validate_foreign_key_options(options)
```

Comment:
could you use the existing `assert_valid_keys` method? http://apidock.com/rails/Hash/assert_valid_keys